code/game/objects/items/rcd/RHD.dm

```diff
@@ -0,0 +1,291 @@
+//RAPID HANDHELD DEVICE. the base for all rapid devices
+
+/obj/item/construction
+	name = "not for ingame use"
+	desc = "A device used to rapidly build and deconstruct. Reload with iron, plasteel, glass or compressed matter cartridges."
+	opacity = FALSE
+	density = FALSE
+	anchored = FALSE
+	flags_1 = CONDUCT_1
+	item_flags = NOBLUDGEON
+	force = 0
+	throwforce = 10
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_NORMAL
+	custom_materials = list(/datum/material/iron=100000)
+	req_access = list(ACCESS_ENGINE_EQUIP)
+	armor_type = /datum/armor/item_construction
+	resistance_flags = FIRE_PROOF
+	var/datum/effect_system/spark_spread/spark_system
+	var/matter = 0
+	var/max_matter = 100
+	var/no_ammo_message = "<span class='warning'>The \'Low Ammo\' light on the device blinks yellow.</span>"
+	var/has_ammobar = FALSE //controls whether or not does update_icon apply ammo indicator overlays
+	var/ammo_sections = 10 //amount of divisions in the ammo indicator overlay/number of ammo indicator states
+	/// Bitflags for upgrades
+	var/upgrade = NONE
+	/// Bitflags for banned upgrades
+	var/banned_upgrades = NONE
+	var/datum/component/remote_materials/silo_mats //remote connection to the silo
+	var/silo_link = FALSE //switch to use internal or remote storage
+
+/datum/armor/item_construction
+	fire = 100
+	acid = 50
+
+/obj/item/construction/Initialize(mapload)
+	. = ..()
+	spark_system = new /datum/effect_system/spark_spread
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+	if(upgrade & RCD_UPGRADE_SILO_LINK)
+		silo_mats = AddComponent(/datum/component/remote_materials, "RCD", mapload, FALSE)
+
+///used for examining the RCD and for its UI
+/obj/item/construction/proc/get_silo_iron()
+	if(silo_link && silo_mats.mat_container && !silo_mats.on_hold())
+		return silo_mats.mat_container.get_material_amount(/datum/material/iron)/500
+	return FALSE
+
+///returns local matter units available. overriden by rcd borg to return power units available
+/obj/item/construction/proc/get_matter(mob/user)
+	return matter
+
+/obj/item/construction/examine(mob/user)
+	. = ..()
+	. += "It currently holds [get_matter(user)]/[max_matter] matter-units."
+	if(upgrade & RCD_UPGRADE_SILO_LINK)
+		. += "Remote storage link state: [silo_link ? "[silo_mats.on_hold() ? "ON HOLD" : "ON"]" : "OFF"]."
+		var/iron = get_silo_iron()
+		if(iron)
+			. += "Remote connection has iron in equivalent to [iron] RCD unit\s." //1 matter for 1 floor tile, as 4 tiles are produced from 1 iron
+
+/obj/item/construction/Destroy()
+	QDEL_NULL(spark_system)
+	silo_mats = null
+	return ..()
+
+/obj/item/construction/pre_attack(atom/target, mob/user, params)
+	if(istype(target, /obj/item/rcd_upgrade))
+		install_upgrade(target, user)
+		return TRUE
+	if(insert_matter(target, user))
+		return TRUE
+	return ..()
+
+/obj/item/construction/attackby(obj/item/W, mob/user, params)
```

Comment:
This would require a whole path replace script throughout for the all places that use rcd,rld  etc. Is there such a thing? Or maybe it's not worth the effort since it works

Reply:
I mean instead of using W

Reply:
Ohhhh. Sure 👍